x-pack/plugins/enterprise_search/server/lib/connectors/put_update_filtering.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
+
+import { CONNECTORS_INDEX } from '../..';
+import {
+  Connector,
+  FilteringRule,
+  FilteringRules,
+  FilteringValidationState,
+} from '../../../common/types/connectors';
+
+import { fetchConnectorById } from './fetch_connectors';
+
+export const updateFiltering = async (
+  client: IScopedClusterClient,
+  connectorId: string,
+  {
+    advancedSnippet,
+    filteringRules,
+  }: {
+    advancedSnippet: string;
+    filteringRules: FilteringRule[];
+  }
+): Promise<FilteringRules | undefined> => {
+  const now = new Date().toISOString();
+  const parsedAdvancedSnippet: Record<string, unknown> = advancedSnippet
+    ? JSON.parse(advancedSnippet)
+    : {};
+  const parsedFilteringRules = filteringRules.map((filteringRule) => ({
+    ...filteringRule,
+    created_at: filteringRule.created_at ? filteringRule.created_at : now,
+    updated_at: now,
+  }));
+  const active: FilteringRules = {
+    advanced_snippet: {
+      created_at: now,
```

Comment:
Should this default to the existing `created_at`?

Reply:
Maybe? But then it'll just always be the same as the date the connector was created, which seems a bit pointless

Reply:
:nods: yep, seeing now that either way, that date field is pointless 💀 :)